test/client/utility.cc

```diff
@@ -16,6 +16,12 @@ std::unique_ptr<OptionsImpl> TestUtility::createOptionsImpl(absl::string_view ar
 }
 
 std::unique_ptr<OptionsImpl> TestUtility::createOptionsImpl(const std::vector<const char*>& argv) {
+  // This works around an error thrown by TCLAP about multiple unlabeled optional args not being
+  // allowed. TCLAP has a global flag that detects multiple unlabeled optional args. It assumes
+  // there will be only one command line in the lifetime of the process. In unit tests we parse
+  // multiple TCLAP command lines, so we need to reset TCLAP's flag to simulate a fresh process.
+  TCLAP::OptionalUnlabeledTracker::alreadyOptional() = false;
```

Comment:
I wonder if there would be anything against doing this in the after we're done parsing TCLAP args in the related `OptionImpl` constructor (plus maybe in the destructor of `OptionImpl` for good measure)..

Reply:
I was torn about this -- putting it in OptionImpl would eliminate the problem for all future tests, but I don't know if we have any rule against modifying production code to fix tests.

Reply:
Test only changes should be avoided... but thinking about it .. arguably this isn't test only. It may be possible to accidentally misuse this? 
 
So maybe, to guard against incidents / foolproof it, a helper class that facilitates enforcing non-concurrent usage as well as taking care of resetting the flag in a RAII-style manner is worth considering here? If feel this would bloat this PR though, it's probably something that should have been done earlier if TCLAP assumes there will be only one command line per process.
Shall we file an issue for this one?

Reply:
Sounds good, filed #259